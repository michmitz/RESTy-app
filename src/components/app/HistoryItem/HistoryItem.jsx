import React from 'react';
import PropTypes from 'prop-types';

const HistoryItem = ({ method, url }) => (
  <div className="history-">
    <p>{method}</p>
    <p>{url}</p>
  </div>
);

HistoryItem.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default HistoryItem;