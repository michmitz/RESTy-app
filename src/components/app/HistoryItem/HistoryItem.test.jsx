import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryItem from './HistoryItem'

describe('Article component', () => {
  afterEach(() => cleanup());
  it('renders history item', () => {
    const { asFragment } = render(<HistoryItem
      method="GET"
      url="test-url.com"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});