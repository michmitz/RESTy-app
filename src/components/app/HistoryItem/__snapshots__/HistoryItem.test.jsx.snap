// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders history item 1`] = `
<DocumentFragment>
  <div
    class="history-item"
  >
    <p>
      GET
    </p>
    <p>
      test-url.com
    </p>
  </div>
</DocumentFragment>
`;
